jwql/instrument_monitors/common_monitors/bad_pixel_monitor.py

```diff
@@ -0,0 +1,1066 @@
+                print('DARKS')
+                for entry in new_dark_entries:
+                    print(entry['detector'], entry['apername'], entry['filter'], entry['readpatt'])
+                #print('FLATS - PREV')
+                #for entry in new_flat_entries_prev:
+                #    print(entry['detector'], entry['apername'], entry['filter'], entry['readpatt'], entry['exp_type'], entry['lamp'], entry['template'])
+
+                # filtering could be different for flats vs darks.
+                # Kevin says we shouldn't need to worry about mixing lamps in the data used to create the bad pixel
+                # mask. In flight, data will only be taken with LINE2, LEVEL 5. Currently in MAST all lamps are
+                # present, but Kevin is not concerned about variations in flat field strucutre.
+                new_flat_entries = self.filter_query_results(new_flat_entries, datatype='flat')
+                new_dark_entries = self.filter_query_results(new_dark_entries, datatype='dark')
+
+                print('FILTERED FLATS')
+                for entry in new_flat_entries:
+                    print(entry['detector'], entry['apername'], entry['filter'], entry['readpatt'], entry['exp_type'], entry['lamp'], entry['template'])
+
+                print('FILTERED DARKS')
+                for entry in new_dark_entries:
+                    print(entry['detector'], entry['apername'], entry['filter'], entry['readpatt'], entry['exp_type'], entry['lamp'], entry['template'])
+
+                # NIRISS - results can include rate, rateints, trapsfilled
+                # MIRI - Jane says they now use illuminated data!!
+                # NIRSpec - can be cal, x1d, rate, rateints can have both cal and x1d so filter repeats
+                # FGS - rate, rateints, trapsfilled
+                # NIRCam - no int flats
+
+                # The query results can contain multiple entries for files
+                # in different calibration states (or for different output
+                # products), so we need to filter the list for duplicate
+                # entries and for the calibration state we are interested
+                # in before we know how many new entries there really are.
+
+                # In the end, we need rate files as well as uncal files
+                # because we're going to need to create jump files.
+                rate_files = []
+                uncal_files = []
+
+                flat_uncal_files = locate_uncal_files(new_flat_entries)
+                dark_uncal_files = locate_uncal_files(new_dark_entries)
+
+                # Remove duplicates and check if there are enough new flat
+                # field files
+                if len(flat_uncal_files) > 0:
+                    flat_uncal_files = sorted(list(set(flat_uncal_files)))
+
+                if len(flat_uncal_files) < flat_file_count_threshold:
```

Comment:
If that were made an `elif`, then if enough files were found to run the monitor, they would go into the `if` block to be sorted and have repeats removed, but then they would not go into the `else` block, which is what calls the monitor itself. The first `if` statement needs to be separate, so that if any files are found, any repeats will be removed, and then the second `if/else` block can decide if there are enough unique files to call the monitor or not.

Yes, I was assuming that the threshold would always be more than 0, since having a threshold of 0 doesn't really make sense.